### DISCUSSION TITLE
Error on program launch:  Could not load file or assembly 'CefSharp.Core.Runtime, Version=90.6.50.0, .....  a program with an incorrect format.

### DISCUSSION THREAD

**Discussion by dmiddents-accesso** [cefsharp/CefSharp#4072] [General]
The full error is 
"Could not load file or assembly 'CefSharp.Core.Runtime, Version=90.6.50.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138'. An attempt was made to load a program with an incorrect format."

I've got this utility running without problem on dozens (hundreds?) of PCs.  It was installed on some new PCs today.  Which are in the UK - it shouldn't make a difference, but is the only thing "special" about this install.  The program won't start and writes the error above to the Windows Event Viewer.

It sounds like I've got a 32bit vs 64bit mismatch.  But if that were the case, the program shouldn't run at all.  I tried coping only 64bit DLLs to the program directory (rather than having the files picked up from the Runtime directory). No difference.  Any ideas?

Thanks,

**Comment by amaitland** [comment]
What version of .Net is your application built using? For .Net Core and .Net 5/6 you'll get this error when Microsoft Visual C ++ is missing from the machine. 

https://docs.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170#visual-studio-2015-2017-2019-and-2022

**Comment by dmiddents-accesso** [comment]
@amaitland - that's exactly what the problem was.  The redistributable was installed, and we are up and running.  Thank you!

**Reply by amaitland** [reply]
Great 👍 Easy fix.

I've added a new FAQ entry at https://github.com/cefsharp/CefSharp/wiki/Frequently-asked-questions#NetCoreBadImageFormatException
